### PR TITLE
Document that load_inline requires having a compiler installed

### PR DESCRIPTION
**Pull request by janeyx99**
Prompted by this forum q: https://discuss.pytorch.org/t/are-the-requirements-for-using-torch-utils-cpp-extension-with-cuda-documented-anywhere/211222

Would be curious to know if we could get more precise.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137521

### CODE REVIEW

File: torch/utils/cpp_extension.py [hunk 1]
```diff
@@ -1629,8 +1629,15 @@ def load_inline(name,
         ...                      functions=['sin_add'])
 
     .. note::
-        By default, the Ninja backend uses #CPUS + 2 workers to build the
-        extension. This may use up too many resources on some systems. One
+        Since load_inline will just-in-time compile the source code, please ensure
+        that you have the right toolchains installed in the runtime. For example,
+        if you are loading C++, make sure a C++ compile is available. If you're loading
```
Comment (zou3519) [inline torch/utils/cpp_extension.py:1634]: nit: compiler

**Review by zou3519 — APPROVED** [review]
Looks good to me but @malfet would know better if we can make this more precise

File: torch/utils/cpp_extension.py [hunk 2]
```diff
@@ -1629,8 +1629,15 @@ def load_inline(name,
         ...                      functions=['sin_add'])
 
     .. note::
-        By default, the Ninja backend uses #CPUS + 2 workers to build the
-        extension. This may use up too many resources on some systems. One
+        Since load_inline will just-in-time compile the source code, please ensure
+        that you have the right toolchains installed in the runtime. For example,
+        if you are loading C++, make sure a C++ compiler is available. If you're loading
```
Comment (albanD) [inline torch/utils/cpp_extension.py:1634]: > if you are loading C++

Why the "if", isn't this always loading c++ code?

File: torch/utils/cpp_extension.py [hunk 3]
```diff
@@ -1629,8 +1629,15 @@ def load_inline(name,
...
+        a CUDA extension, you will need to additionally install the corresponding CUDA nvcc
+        and any other dependencies your code has. Compiling toolchains are not included
```
Comment (albanD) [inline torch/utils/cpp_extension.py:1636]: nit: I would name this "cuda toolkit (that contains nvcc and necessary headers in particular)

Comment (janeyx99) [reply torch/utils/cpp_extension.py:1636]: Is this cuda toolkit the same one available on conda? Cuz the forum q poster mentioned installing this and having it not be enough

File: torch/utils/cpp_extension.py (hunk 2 above)
Comment (janeyx99) [reply torch/utils/cpp_extension.py:1634]: okay yea

**Pull request closed without merging**